contracts/Vault.vy

```diff
@@ -1100,8 +1110,10 @@ def withdraw(
                 continue  # Nothing to withdraw from this Strategy, try the next one
 
             # Force withdraw amount from each Strategy in the order set by governance
+            preBalance: uint256 = self.token.balanceOf(self)
             loss: uint256 = Strategy(strategy).withdraw(amountNeeded)
```

Comment:
I was looking for edge cases on loss > amountNeeded and potentially eating the difference between idle and self.token.balanceOf(self), but this math seems to be right

Also remember we have many tests cases around reporting losses and accounting so if we have no regression on those should be good sign